### ISSUE TITLE
logging: WARN messages on every agent start

### ISSUE THREAD

**Issue by shoenig** [hashicorp/nomad#16288]
Lately I've been seeing these WARN level agent log messages on every agent startup. Either something is actually wrong and we should investigate, or these should not be warnings.

```
2023-03-01T10:29:42.765-0600 [WARN]  client.alloc_runner.task_runner.task_hook.logmon: plugin configured with a nil SecureConfig: alloc_id=828b90e5-1fa6-2a31-9acb-dafe46a90d9b task=task
2023-03-01T10:29:43.058-0600 [WARN]  client.driver_mgr.docker.docker_logger: plugin configured with a nil SecureConfig: driver=docker
```

**Comment by tgross** [comment]
Looks like this came in from [go-plugin v1.4.5](https://github.com/hashicorp/go-plugin/blob/main/CHANGELOG.md#v145), which was introduced by this dependabot commit: https://github.com/hashicorp/nomad/commit/8b77a419bbb9c641553f9e3e7a0b96fefa31022d

**Comment by tgross** [comment]
So `go-plugin` [#207](https://github.com/hashicorp/go-plugin/pull/207) introduced a warning if the `SecureConfig` field is unset. For Nomad this means that all the "internal" plugins where Nomad is running itself as a plugin (task drivers and logmon) are spamming warning-level logs. I tried swapping this out for a non-nil `SecureConfig` but then the plugins fail checksumming because the checksum (which is a hash of the binary) is unset. I've opened an internal discussion on Slack to see what other folks with this architecture are doing for this problem.
